syntax/scan.go

```diff
@@ -276,6 +276,7 @@ func readSource(filename string, src interface{}) ([]byte, error) {
 		data, err := ioutil.ReadAll(src)
 		if err != nil {
 			err = &os.PathError{Op: "read", Path: filename, Err: err}
+			return nil, err
```

Comment:
Good catch. I suspect no-one ever passes a Reader here which is why it went unnoticed.